lightning/src/util/ser.rs

```diff
@@ -592,7 +596,9 @@ impl<T: Writeable> Writeable for Option<T> {
 		match *self {
 			None => 0u8.write(w)?,
 			Some(ref data) => {
-				1u8.write(w)?;
+				let mut len_calc = LengthCalculatingWriter(0);
```

Comment:
nit: is commit message right here? "Change Option<T>::read serialization format to include length" you also change Option<T>::write